srcpkgs/cmake/template

```diff
@@ -15,14 +18,32 @@ maintainer="Đoàn Trần Công Danh <congdanhqx@gmail.com>"
 license="BSD-3-Clause, ICU"
 homepage="https://www.cmake.org"
 distfiles="https://www.cmake.org/files/v${version%.*}/${pkgname}-${version}.tar.gz"
-checksum=4d008ac3461e271fcfac26a05936f77fc7ab64402156fb371d41284851a651b8
+checksum=12c8040ef5c6f1bc5b8868cede16bb7926c18980f59779e299ab52cbc6f15bb0
 replaces="cmake-bootstrap>=0"
 
-if [ "$CROSS_BUILD" ]; then
-	build_style=cmake
-	configure_args="-DCMAKE_DOC_DIR=/share/doc/cmake
-	 -DCMAKE_USE_SYSTEM_LIBRARIES=ON -DKWSYS_LFS_WORKS=1"
-fi
+# XXX: cmake is broken if cmake was built with -GNinja
+# https://bugs.gentoo.org/596460
+export CMAKE_GENERATOR="Unix Makefiles"
+make_cmd=make
+
+pre_configure() {
+	local f
+	mkdir -p $cmake_builddir
+	cd $cmake_builddir
+	CC="$CC_FOR_BUILD" CFLAGS="$CFLAGS_FOR_BUILD" \
+	CXX="$CXX_FOR_BUILD" CXXFLAGS="$CXXFLAGS_FOR_BUILD" \
+	LD="$LD_FOR_BUILD" LDFLAGS="$LDFLAGS_FOR_BUILD" \
+	../bootstrap --no-system-libs \
+		${XBPS_MAKEJOBS:+--parallel=$XBPS_MAKEJOBS}
```

Comment:
Why bootstrap a native cmake for every build instead of depending on `cmake` for cross builds? Not sure I understand the advantage. This feels more complicated to maintain and would take longer to build.

Reply:
It only compiles 276 files. On cross-compiling, if we depends on `cmake` in `hostmakedepends`. Sometime, when host `cmake` is being built, the cross-builder will build host `cmake` themselves, thus wasting effort.